### PR TITLE
feat(Makefile): enable immutable (git-based) Docker tag publishing

### PR DESCRIPTION
**Pull request by sgoings**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -49,27 +44,12 @@ test:
...
-kube-service:
-	kubectl create -f ${SVC}
-
-# When possible, we deploy with RCs.
-kube-rc:
-	kubectl create -f ${RC}
-
-kube-clean:
-	kubectl delete rc deis-${SHORT_NAME}
-	kubectl delete svc deis-${SHORT_NAME}
```
Comment (sgoings) [inline Makefile]: These targets were unusable to me (since we're not storing manifests in the repo anymore).

Comment (helgi) [inline Makefile]: You could port over `make deploy` from `controller`

Comment (arschles) [inline Makefile]: :+1: on deleting these targets here, but @sgoings or @helgi can you create an issue in this repo for porting `make deploy` into here?

Comment (sgoings) [inline Makefile]: @arschles: #249

**Comment by bacongobbler** [comment]
ping @sgoings; your PR is good to go. :shipit:

**Pull request merged**